logstash-core/lib/logstash/patches/manticore/trust_strategies.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+
+require 'manticore'
+
+unless defined?(::Manticore::Client::TrustStrategies)
+  module CATrustedFingerprintSupport
+    # @override
+    def setup_trust_store(ssl_options, context, trust_strategy)
+      if ssl_options.include?(:trust_strategy)
+        trust_strategy = TrustStrategies.combine(trust_strategy, ssl_options.fetch(:trust_strategy))
+      end
+
+      super(ssl_options, context, trust_strategy)
+    end
+
+    module TrustStrategies
+      def self.combine(lhs, rhs)
+        return rhs if lhs.nil?
+        return lhs if rhs.nil?
+
+        Multiple.new([lhs, rhs])
+      end
+
+      class Multiple
+        include ::Java::OrgApacheHttpConnSSL::TrustStrategy
+
+        def initialize(trust_strategies)
+          @trust_strategies = trust_strategies.compact
+          super()
+        end
+
+        # boolean isTrusted(X509Certificate[] chain, String authType) throws CertificateException;
+        def trusted?(x509_certificate_chain, auth_type)
+          @trust_strategies.any? do |trust_strategy|
+            trust_strategy.trusted?(x509_certificate_chain, auth_type)
+          end
```

Comment:
More closely mirror the combined trust strategy implemented upstream. Note that we don't have to deal with wrapping like the upstream implementation because we are exclusively dealing with instances of `org.apache.http.conn.ssl.TrustStrategy` whereas upstream also has a handler for `Proc` that uses ruby types.

```suggestion
        Combined.new(lhs, rhs)
      end

      class Combined
        include ::Java::OrgApacheHttpConnSSL::TrustStrategy

        def initialize(lhs, rhs)
          @lhs = lhs
          @rhs = rhs
          super()
        end

        # boolean isTrusted(X509Certificate[] chain, String authType) throws CertificateException;
        def trusted?(x509_certificate_chain, auth_type)
          return true if @lhs.trusted?(x509_certificate_chain, auth_type)
          return true if @rhs.trusted?(x509_certificate_chain, auth_type)
          
          false
```